patches/minecraft/net/minecraft/potion/Potion.java.patch

```diff
@@ -88,6 +88,18 @@
 +        ret.add(new net.minecraft.item.ItemStack(net.minecraft.init.Items.field_151117_aB));
 +        return ret;
 +    }
++    
++    /**
++     * Return an index that will override the potion's default sort order. This is useful for potions with colors 
++     * that change since, without this, the effect order would shift in the inventory GUI when stacked with others.
++     * 
++     * @param The PotionEffect instance containing the potion
++     * @return An index to override the default sort order
++     */
++    public int getSortIndex(PotionEffect potionEffect)
```

Comment:
"index" doesn't feel like the right word to use here. Maybe just "sortValue" or even "sortOverrideValue." Or maybe "index" really *is* the best choice.